src/process_utility.c

```diff
@@ -882,7 +914,142 @@ timescaledb_ProcessUtility(Node *parsetree,
 			break;
 	}
 
-	prev_ProcessUtility(parsetree, query_string, context, params, dest, completion_tag);
+	return handled;
+}
+
+/*
+ * Handle DDL commands after they've been processed by PostgreSQL.
+ */
+static void
+process_ddl_command_end(CollectedCommand *cmd)
+{
+	switch (nodeTag(cmd->parsetree))
+	{
+		case T_AlterTableStmt:
+			process_altertable(cmd->parsetree, cmd);
+			break;
+		case T_CreateTrigStmt:
+			process_create_trigger(cmd->parsetree);
+			break;
+		default:
+			break;
+	}
+}
+
+/*
+ * ProcessUtility hook for DDL commands that have not yet been processed by
+ * PostgreSQL.
+ */
+static void
+timescaledb_ddl_command_start(Node *parsetree,
+							  const char *query_string,
+							  ProcessUtilityContext context,
+							  ParamListInfo params,
+							  DestReceiver *dest,
+							  char *completion_tag)
+{
+	/*
+	 * If we are restoring, we don't want to recurse to chunks or block
+	 * operations on chunks. If we do, the restore will fail.
+	 */
+	if (!extension_is_loaded() || guc_restoring)
+	{
+		prev_ProcessUtility(parsetree, query_string, context, params, dest, completion_tag);
+		return;
+	}
+
+	if (!process_ddl_command_start(parsetree, query_string, context, completion_tag))
+		prev_ProcessUtility(parsetree, query_string, context, params, dest, completion_tag);
+}
+
+/* Function manager info for the event "pg_event_trigger_ddl_commands", which is
+ * used to retrieve information on executed DDL commands in an event
+ * trigger. The function manager info is initialized on extension load. */
+static FmgrInfo ddl_commands_fmgrinfo;
+
+/*
+ * Get a list of executed DDL commands in an event trigger.
+ *
+ * This function calls the function pg_event_trigger_ddl_commands(), which is
+ * part of the event trigger API, and retrieves the DDL commands executed in
+ * relation to the event trigger. It is only valid to call this function from
+ * within an event trigger.
+ */
+static List *
+event_trigger_ddl_commands()
```

Comment:
Should all the event trigger stuff be in a separate file? Kind of awkward for it to be in a process_utility.c?

Reply:
I see why it's one file now. So kind of mixed on what we should do here. I'll leave it up to you.